cocotb/triggers.py

```diff
@@ -934,7 +940,12 @@ def __repr__(self):
         return fmt.format(type(self).__qualname__, self.signal, self.num_cycles)
 
 
-async def with_timeout(trigger, timeout_time, timeout_unit="step"):
+async def with_timeout(
+    trigger: Union[Trigger, Waitable, Coroutine[Any, Any, T]],
```

Comment:
The reason that `cocotb.coroutine` is not listed here is because of a circular dependency =/ We need to make a clear dependency graph.

Reply:
Closing #3067 would solve this.

Reply:
Usually you can write `cocotb.coroutine` when that happens